File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.cs
public class Solution {
    public int[] TopKFrequent(int[] nums, int k) {
        Dictionary< int , int > numberFrequency = new Dictionary< int , int>();
        
        foreach( int n in nums){
            if(!numberFrequency.ContainsKey(n)){
                numberFrequency[n]=1;
            }
            else{
                numberFrequency[n]++;
            }
        }

        int [] topKElements = numberFrequency
            .OrderByDescending(pair => pair.Value)
            .Take(k)
            .Select(pair => pair.Key )
            .ToArray();
        
        return topKElements;


    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-4.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        nums_set : set[int] = set()

        for n in nums:

            if n not in nums_set: 
                nums_set.add(n)
            

            else:
                return True
            

        return False